src/marqo_commons/model_registry/model_properties_object.py

```diff
@@ -23,6 +25,11 @@ def __init__(self, **kwargs):
             kwargs["memory_size"] = self._get_default_model_size(kwargs["name"])
         super().__init__(**kwargs)
 
+    @classmethod
+    @abstractmethod
+    def list_model_properties(cls) -> Dict[str, T]:
```

Comment:
Add a docstring explaining this returns ModelProperties objects. It is separate from the other methods like `_get_sbert_properties` which return dicts